app/models/intake/ctc_intake.rb

```diff
@@ -287,6 +287,18 @@ class Intake::CtcIntake < Intake
     self.spouse_ssn = self.spouse_ssn.remove(/\D/) if spouse_ssn_changed? && self.spouse_ssn
   end
 
+  before_save do
+    attributes_to_change = self.changes_to_save.keys
+    name_attributes = ["primary_first_name", "primary_last_name", "spouse_first_name", "spouse_last_name"]
+
+    (attributes_to_change & name_attributes).each do |attribute|
+      if self.send(attribute).present?
```

Comment:
is the presence check here to guard against nil? because seems like attributes_to_change is already filtering to present attributes. 

_If_ the reason is to prevent against running this on attributes that are blank or nil, maybe `unless self.send(attribute).blank?` might be more clear, not sure.

Reply:
Can this be `self.attributes[attribute]` rather than `self.send(attribute)`? I have a preference for minimizing use of `send` if something else will do the job. Not a blocker.